docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
In the most common case, you're using an ephemeral private key here, generated specifically for this exchange. It's very unlikely that you would ever want to use the key with more than one peer, and you really don't want to keep the key around once the shared key has been computed as that might allow an attacker to more easily figure out the shared key (since they only need that private key and the peer's public key that is often sent in the clear).

My ideal API here would actually build that in, generating the private key at random when the key exchange object was created, and providing APIs to return the public key to send to the peer and to generate the shared key from the peer's public key.